vocab/security/vocabulary.yml

```
@@ -272,6 +272,12 @@ property:
     range: xsd:string
     comment: The challenge property is used to associate a challenge with a proof, for use with a `proofPurpose` such as `authentication`. This string value SHOULD be included in a proof if a `domain` is specified.
 
+  - id: previousProof
+    label: Previous proof
+    domain: sec:Proof
+    range: xsd:anyURI
```

```suggestion
    range: sec:Proof
```

My understanding, per description, is that we refer to another proof that must be checked before the current one. Ie, the object of the property must be a proof as well, right?

The minor downside is that this change makes it possible to create "embedded" proofs, ie, that the previous proof is in a blank node. I do not think it is actually a real problem but if it is, this can be explicitly restricted in the description within the specification.

---

> My understanding, per description, is that we refer to another proof that must be checked before the current one. Ie, the object of the property must be a proof as well, right?
> 
> The minor downside is that this change makes it possible to create "embedded" proofs, ie, that the previous proof is in a blank node. I do not think it is actually a real problem but if it is, this can be explicitly restricted in the description within the specification.

i am good with this adjustment as noted by @iherman - marking approved for now, as I think we can open an issue and address, even if we take this PR as is (since it is a clear improvement)

---

> My understanding, per description, is that we refer to another proof that must be checked before the current one. Ie, the object of the property must be a proof as well, right?

Yes, and I was trying to keep things simple so that we always deal w/ references and a flat proof array, and not a mixture of embedded proofs and references to proofs.

We can say `sec:Proof` and restrict the property in the spec to a URL (which I think it already is).